src/main/java/de/plushnikov/intellij/plugin/processor/clazz/log/AbstractLogProcessor.java

```diff
@@ -94,22 +110,44 @@ private LombokLightFieldBuilder createLoggerField(@NotNull PsiClass psiClass, @N
       loggerField.withModifier(PsiModifier.STATIC);
     }
 
-    final String loggerInitializerParameter = createLoggerInitializeParameter(psiClass, psiAnnotation);
-    final PsiExpression initializer = psiElementFactory.createExpressionFromText(String.format(loggerInitializer, loggerInitializerParameter), psiClass);
+    final String loggerInitializerParameters = createLoggerInitializeParameters(psiClass, psiAnnotation);
+    final String initializerText = String.format(getLoggerInitializer(psiClass), loggerInitializerParameters);
+    final PsiExpression initializer = psiElementFactory.createExpressionFromText(initializerText, psiClass);
     loggerField.setInitializer(initializer);
     return loggerField;
   }
 
   @NotNull
-  private String createLoggerInitializeParameter(@NotNull PsiClass psiClass, @NotNull PsiAnnotation psiAnnotation) {
-    final String loggerInitializerParameter;
+  private String createLoggerInitializeParameters(@NotNull PsiClass psiClass, @NotNull PsiAnnotation psiAnnotation) {
+    final StringBuilder parametersBuilder = new StringBuilder();
     final String topic = PsiAnnotationUtil.getStringAnnotationValue(psiAnnotation, "topic");
-    if (StringUtil.isEmptyOrSpaces(topic)) {
-      loggerInitializerParameter = String.format(loggerCategory, psiClass.getName());
-    } else {
-      loggerInitializerParameter = '"' + topic + '"';
+    final boolean topicPresent = !StringUtil.isEmptyOrSpaces(topic);
+    final List<LoggerInitializerParameter> loggerInitializerParameters = getLoggerInitializerParameters(psiClass, topicPresent);
+    for (LoggerInitializerParameter loggerInitializerParameter : loggerInitializerParameters) {
+      if (parametersBuilder.length() > 0) {
+        parametersBuilder.append(", ");
+      }
+      switch(loggerInitializerParameter) {
+        case TYPE:
+          parametersBuilder.append(psiClass.getName()).append(".class");
+          break;
+        case NAME:
+          parametersBuilder.append(psiClass.getName()).append(".class.getName()");
+          break;
+        case TOPIC:
+          if (!topicPresent) {
+            throw new IllegalStateException("Topic can never be a parameter when topic was not set.");
```

Comment:
Please don't throw any exceptions during augment generation. Let checks like this be a part of validation step and just not start generation step.

Reply:
This exception cannot happen; it is just a sanity check.